eks/fluentd/tester.go

```diff
@@ -0,0 +1,184 @@
+}
+
+type tester struct {
+	cfg Config
+
+	busyboxImg string
+}
+
+func (ts *tester) Create() (err error) {
+	if !ts.cfg.EKSConfig.IsEnabledAddOnFluentd() {
+		ts.cfg.Logger.Info("skipping tester.Create", zap.String("tester", pkgName))
+		return nil
+	}
+	if ts.cfg.EKSConfig.AddOnFluentd.Created {
+		ts.cfg.Logger.Info("skipping tester.Create", zap.String("tester", pkgName))
+		return nil
+	}
+
+	ts.cfg.Logger.Info("starting tester.Create", zap.String("tester", pkgName))
+	ts.cfg.EKSConfig.AddOnFluentd.Created = true
+	ts.cfg.EKSConfig.Sync()
+	createStart := time.Now()
+	defer func() {
+		createEnd := time.Now()
+		ts.cfg.EKSConfig.AddOnFluentd.TimeFrameCreate = timeutil.NewTimeFrame(createStart, createEnd)
+		ts.cfg.EKSConfig.Sync()
+	}()
+
+	if ts.cfg.EKSConfig.AddOnFluentd.RepositoryBusyboxAccountID != "" &&
+		ts.cfg.EKSConfig.AddOnFluentd.RepositoryBusyboxRegion != "" &&
+		ts.cfg.EKSConfig.AddOnFluentd.RepositoryBusyboxName != "" &&
+		ts.cfg.EKSConfig.AddOnFluentd.RepositoryBusyboxImageTag != "" {
+		if ts.busyboxImg, err = aws_ecr.Check(
+			ts.cfg.Logger,
+			ts.cfg.ECRAPI,
+			ts.cfg.EKSConfig.AddOnFluentd.RepositoryBusyboxAccountID,
+			ts.cfg.EKSConfig.AddOnFluentd.RepositoryBusyboxRegion,
+			ts.cfg.EKSConfig.AddOnFluentd.RepositoryBusyboxName,
+			ts.cfg.EKSConfig.AddOnFluentd.RepositoryBusyboxImageTag,
+		); err != nil {
+			return err
+		}
+	}
+
+	if err = k8s_client.CreateNamespace(
+		ts.cfg.Logger,
+		ts.cfg.K8SClient.KubernetesClientSet(),
+		ts.cfg.EKSConfig.AddOnFluentd.Namespace,
+	); err != nil {
+		return err
+	}
+
+	// create Fluentd components
+	if err = ts.createFluentdServiceAccount(); err != nil {
+		return err
+	}
+	if err = ts.createFluentdRBACClusterRole(); err != nil {
+		return err
+	}
+	if err = ts.createFluentdRBACClusterRoleBinding(); err != nil {
+		return err
+	}
+	if err = ts.createFluentdConfigMapClusterInfo(); err != nil {
+		return err
+	}
+	if err = ts.createFluentdConfigMapConfig(); err != nil {
+		return err
+	}
+	if err = ts.createFluentdDaemonSet(); err != nil {
+		return err
+	}
+	if err = ts.checkFluentdPods(); err != nil {
+		return err
+	}
+
+	return ts.cfg.EKSConfig.Sync()
+}
+
+func (ts *tester) Delete() error {
+	if !ts.cfg.EKSConfig.IsEnabledAddOnFluentd() {
+		ts.cfg.Logger.Info("skipping tester.Delete", zap.String("tester", pkgName))
+		return nil
+	}
+	if !ts.cfg.EKSConfig.AddOnFluentd.Created {
+		ts.cfg.Logger.Info("skipping tester.Delete", zap.String("tester", pkgName))
+		return nil
+	}
+
+	ts.cfg.Logger.Info("starting tester.Delete", zap.String("tester", pkgName))
+	deleteStart := time.Now()
+	defer func() {
+		deleteEnd := time.Now()
+		ts.cfg.EKSConfig.AddOnFluentd.TimeFrameDelete = timeutil.NewTimeFrame(deleteStart, deleteEnd)
+		ts.cfg.EKSConfig.Sync()
+	}()
+
+	var errs []string
+
+	if err := ts.deleteFluentdDaemonSet(); err != nil {
+		errs = append(errs, err.Error())
+	}
+	time.Sleep(time.Minute)
+
+	if err := ts.deleteFluentdConfigMapConfig(); err != nil {
```

Comment:
You could probably collapse a lot of this code by doing a for loop over a list of these functions

for _, func : range [ts. deleteFluentdConfigMapClusterInfo, deleteFluentdRBACClusterRoleBinding, ...]